bin/hdfcoinc/pycbc_multiifo_combine_statmap

```diff
@@ -0,0 +1,220 @@
+    for sk in fi[k].keys():
+        if isinstance(fi[k+'/'+sk], h5py.Dataset):
+            subkey_list.append(k+'/'+sk)
+        else:
+            subkey_list += get_all_subkeys(fi,k+'/'+sk)
+    # this will return an empty list if there is no dataset or subgroup within the group
+    return subkey_list
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--version", action="version", version=pycbc.version.git_verbose_msg)
+parser.add_argument('--verbose', action='store_true')
+parser.add_argument('--statmap-files', nargs='+',
+                    help="List of coinc files to be redistributed")
+parser.add_argument('--cluster-window', type=float)
+parser.add_argument('--output-file', help="name of output file")
+parser.add_argument('--ifos',nargs='+', 
+                    help="list of interferometers in the input files - note that this " \
+                         "needs to be in the order of pivot/fixed precedence" )
+args = parser.parse_args()
+
+pycbc.init_logging(args.verbose)
+
+files = [h5py.File(n,'r') for n in args.statmap_files]
+
+# copy over the standard data that is constant
+#TODO: This is no longer constant between files
+f = h5py.File(args.output_file, "w")
+
+f.attrs['pivot'] = files[0].attrs['pivot']
+f.attrs['fixed'] = files[0].attrs['fixed']
+f.attrs['timeslide_interval'] = files[0].attrs['timeslide_interval']
+
+f.attrs['background_time'] = files[0].attrs['background_time']
+f.attrs['foreground_time'] = files[0].attrs['foreground_time']
+f.attrs['background_time_exc'] = files[0].attrs['background_time_exc']
+f.attrs['foreground_time_exc'] = files[0].attrs['foreground_time_exc']
+
+
+# combine times that are foreground vetoed as they may differ between bins
+
+for fi in files:
+    ifos_in_file = [ifo for ifo in args.ifos if ifo in fi['segments']]
+    ifo_to_get = 'segments/{}'.format(ifos_in_file[0])
+    group_to_save = 'segments/{}'.format(''.join(ifos_in_file))
+    f[group_to_save + '/end'] = fi[ifo_to_get + '/end'][:]
+    f[group_to_save + '/start'] = fi[ifo_to_get + '/start'][:]
+
+key_set = name_all_datasets(files)
+# copy and concatenate all the columns in the foreground group 
+# from all files /except/ the IFO groups, we recalculate ifar/fap later
+for key in key_set:
+    if key.startswith('foreground') and not any([ifo in key for ifo in args.ifos]):
+        pycbc.io.combine_and_copy(f, files, key)
+
+
+#for the ifo-specific groups, copy them over into the new file
+# - if the ifo is not present for each coincident trigger, 
+#  indicate by including a -1 in the combined file
+
+trigger_list = []
+all_trig_times = {}
+all_trig_ids = {}
+for ifo in args.ifos:
+    all_trig_times[ifo] = numpy.array([], dtype=numpy.uint32)
+    all_trig_ids[ifo] = numpy.array([], dtype=numpy.uint32)
+    # for each file, append the trigger time and id data for each ifo
+    #   - if it isnt there, then fill with -1 values
+for f_in in files:
+    for ifo in args.ifos:
+        if ifo in ifo in f_in['foreground']:
```

Comment:
I can't understand this line ...

Reply:
Neither can I.
It's a mistype. I dont know how this didnt cause the code to fail